unified_planning/io/python_writer.py

```diff
@@ -268,11 +273,30 @@ def write_problem_code_to_file(self, filename: str):
             self._write_problem_code(f)
 
 
-def _print_python_type(type: 'up.model.types.Type') -> str:
+def _get_mangled_name(original_name: str, names_mapping: Dict[str, str]) -> str:
+    '''Important note: This method updates the nammes_mapping '''
+    new_name: Optional[str] = names_mapping.get(original_name, None)
+    if new_name is None: # The name is not in the map, so it must be added
+        if not original_name.isidentifier(): # Make the name a python identifier
+            new_name = re.sub('[^0-9a-zA-Z_]', '_', f'{original_name}')
```

Comment:
Unfortunately, this is not enough. Try to add a test with an action or fluent called `4theglory` or even just `5`

Reply:
Actually this should not be a problem, because we always "'type_name'_" before the actual name, so it would result in "fluent_4theglory". But to make sure I will add a specific test; better safe than sorry

Reply:
Ah, you are right. Just to be sure, add an assertion that original_name always starts with a `[a-zA-Z_]` (it is the case now, but failing to meet this constraint would beak this function and cause assertion error in line 292)